### PR TITLE
Add basic explanaition of app's extensions

### PR DESCRIPTION
**Pull request by korycins**
It adds more context to how App's extensions can be used.

### CODE REVIEW

File: docs/developer/extending/apps/extensions.mdx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      }
+    ],
+}
+```
+
+- `label` - Name which will be displayed in dashboard
+- `view` - View where the given extension will be attached
+- `type` - Place where the given extension will be attached
+- `target` - Direct target on given view&type where extension will be attached
+- `openAs` - Way of opening an extension's url. Default to `POPUP`.
```
Comment (krzysztofwolski) [inline docs/developer/extending/apps/extensions.mdx:48]: Please add dots at the end of all items or remove.

File: docs/developer/extending/apps/manifest.mdx [hunk 2]
```diff
@@ -37,4 +61,5 @@ Saleor expects the manifest to use the following format:
 - `dataPrivacy`: Short description of privacy policy displayed in the dashboard
 - `dataPrivacyUrl`: URL to the full privacy policy
 - `homepageUrl`: External URL to the app homepage
-- `supportUrl`: External URL to the page where app users can find support 
\ No newline at end of file
+- `supportUrl`: External URL to the page where app users can find support
+- `extensions`: List of extensions that will be mounted in Saleor's dashboard. See [extension section](developer/extending/apps/extensions.mdx#key-concepts).
```
Comment (krzysztofwolski) [inline docs/developer/extending/apps/manifest.mdx:65]:
```suggestion
- `extensions`: List of extensions that will be mounted in Saleor's dashboard. See [extension section](developer/extending/apps/extensions.mdx#key-concepts)
```

**Review by krzysztofwolski — CHANGES_REQUESTED** [review]
Content looks ok

File: docs/developer/extending/apps/extensions.mdx [hunk 3]
```diff
@@ -0,0 +1,68 @@
+---
+title: Extensions
+---
+
+## Key concepts
+
+App extension allows extending different dashboard views by mounting extension iframe or by redirecting user on main App's iframe.
```
Comment (jwm0) [inline docs/developer/extending/apps/extensions.mdx:7]:
```suggestion
App extensions allow extending different dashboard views by mounting the extension app inside a popup or redirecting users to the main app's page. The app is created as a secure iframe that protects against XSS attacks. 
```

File: docs/developer/extending/apps/extensions.mdx [hunk 4]
```diff
@@ -0,0 +1,68 @@
+---
+title: Extensions
+---
+
+## Key concepts
+
+App extension allows extending different dashboard views by mounting extension iframe or by redirecting user on main App's iframe. 
+This is a way for customizing and extending dashboard solutions.
```
Comment (jwm0) [inline docs/developer/extending/apps/extensions.mdx:8]:
```suggestion
It provides a way for customizing and extending the dashboard while keeping the code separate.
```

File: docs/developer/extending/apps/extensions.mdx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+## Manifest
+App can have multiple extensions. The extension is added by specifying it in app's manifest. 
+Below sample manifest, defines two extensions that will be attached to dashboard.
```
Comment (jwm0) [inline docs/developer/extending/apps/extensions.mdx:12]:
```suggestion
Apps can have multiple extensions. The extension is added by specifying it in the app's manifest. 
The below sample manifest defines two extensions that will be attached to the dashboard:
```

File: docs/developer/extending/apps/extensions.mdx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        "permissions": [
+            "MANAGE_PRODUCTS"
+        ],
+        "url": "/extension/redirect"
+      }
+    ],
+}
+```
+
+- `label` - Name which will be displayed in dashboard
```
Comment (jwm0) [inline docs/developer/extending/apps/extensions.mdx:44]:
```suggestion
- `label` - Name which will be displayed in the dashboard
```

File: docs/developer/extending/apps/extensions.mdx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+- `view` - View where the given extension will be attached
+- `type` - Place where the given extension will be attached
+- `target` - Direct target on given view&type where extension will be attached
```
Comment (jwm0) [inline docs/developer/extending/apps/extensions.mdx:47]:
```suggestion
- `target` - Direct target on given view & type where extension will be attached
```

File: docs/developer/extending/apps/extensions.mdx [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+- `openAs` - Way of opening an extension's url. Default to `POPUP`. 
+`POPUP` will mount extension in iframe view. `APP_PAGE` will redirect user to app page on dashboard side.
```
Comment (jwm0) [inline docs/developer/extending/apps/extensions.mdx:49]:
```suggestion
`POPUP` will mount extension in an iframe view. `APP_PAGE` will redirect users to the dashboard's app page.
```

File: docs/developer/extending/apps/manifest.mdx [hunk 9]
```diff
@@ -22,7 +22,27 @@ Saleor expects the manifest to use the following format:
   "dataPrivacy": "Lorem ipsum",
   "dataPrivacyUrl": "http://localhost:3000/app-data-privacy",
   "homepageUrl": "http://localhost:3000/homepage",
-  "supportUrl": "http://localhost:3000/support"
+  "supportUrl": "http://localhost:3000/support",
+  "extensions": [
+      {
+        "label": "Create with Sample app",
+        "mount": "PRODUCT_OVERVIEW_CREATE",
```
[the remark this answers is not in the document]
Comment (korycins) [reply docs/developer/extending/apps/manifest.mdx:29]: I've talked with @jwm0 and we agree that for now, we will leave `mount` as a str. In the future we will be able to extend `mount` to support `list` and `str`

File: docs/developer/extending/apps/extensions.mdx [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+- `target` - Way of opening an extension's url. Default to `POPUP`.
+`POPUP` will mount extension in an iframe view. `APP_PAGE` will redirect users to the dashboard's app page
+- `url` - Extension url. Url can start with `/` when `target` is set to `APP_PAGE` or manifest has defined `appUrl`. 
+When `target` is set to `POPUP`, `url` will be used to render an iframe
+
+
+
+## Possible mounting places
+
+Saleor requires that extension will define place of mounting. The below table explains all mounting places available in Saleor.
```
Comment (karolkielecki) [inline docs/developer/extending/apps/extensions.mdx:52]:
```suggestion
Saleor requires that extension will define a place of mounting. The below table explains all mounting places available in Saleor.
```

File: docs/developer/extending/apps/manifest.mdx [hunk 11]
```diff
@@ -37,4 +57,5 @@ Saleor expects the manifest to use the following format:
 - `dataPrivacy`: Short description of privacy policy displayed in the dashboard
 - `dataPrivacyUrl`: URL to the full privacy policy
 - `homepageUrl`: External URL to the app homepage
-- `supportUrl`: External URL to the page where app users can find support 
\ No newline at end of file
+- `supportUrl`: External URL to the page where app users can find support
+- `extensions`: List of extensions that will be mounted in Saleor's dashboard. See [extension section](developer/extending/apps/extensions.mdx#key-concepts)
```
Comment (karolkielecki) [inline docs/developer/extending/apps/manifest.mdx:61]:
```suggestion
- `extensions`: List of extensions that will be mounted in Saleor's dashboard. For details, please see the [extension section](developer/extending/apps/extensions.mdx#key-concepts)
```

**Review by karolkielecki — CHANGES_REQUESTED** [review]

**Review by karolkielecki — APPROVED** [review]

**Review by krzysztofwolski — APPROVED** [review]

**Pull request merged**